### PR TITLE
Add Dockerfile and configuration files

### PR DESCRIPTION
**Pull request by stephdl**
This pull request adds the necessary files for building and running the application in a Docker container. It includes the Dockerfile, Apache configuration file, cronie configuration file, event listener configuration file, event listener script, httpd configuration, sogod.ini configuration file, and memcached configuration file. It also updates the readme file and adds a license file. Additionally, it removes unused files and configurations.

https://github.com/NethServer/dev/issues/6804

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -81,7 +81,7 @@ works, such as semiconductor masks.
 License.  Each licensee is addressed as "you".  "Licensees" and
 "recipients" may be individuals or organizations.
 
-  To "modify" a work means to copy from or adapt all or part of the work
+  To "modify" a work means tocopy from or adapt all or part of the work
```
Comment (DavidePrincipi) [inline LICENSE]: :grey_question:

File: build-images.sh [hunk 2]
```diff
@@ -7,56 +7,89 @@
 
 # Terminate on error
 set -e
-
+archlinux_version=latest
```
Comment (DavidePrincipi) [inline build-images.sh:10]: I'd prefer pointing to a fixed release number. The renovate bot can monitor the upstream updates.

Comment (stephdl) [reply build-images.sh:10]: hum this is a good point to talk, we are under a rolling release and archlinux itself recommend to start by an update of the image before to install your needs, so if we do this, we could lost a lot of time doing the image update https://hub.docker.com/_/archlinux

Updating
Arch Linux is a rolling release distribution, so a full update is recommended when installing new packages. In other words, we suggest to either execute RUN pacman -Syu immediately after your FROM statement or as soon as you docker run into a container.

File: LICENSE (hunk 1 above)
Comment (stephdl) [reply LICENSE]: sorry I do not understand, you want we remove the License ?

File: build-images.sh (hunk 2 above)
Comment (stephdl) [reply build-images.sh:10]: if we want a target, this is a good one : base-20231112.0.191179

File: LICENSE (hunk 1 above)
Comment (DavidePrincipi) [reply LICENSE]: why did you modify that line?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+WORKDIR /build/libwbxml
+RUN sudo -u build makepkg -is --noconfirm && rm -rf /build/libwbxml && yes | pacman -Sccq
+
+WORKDIR /build
+RUN git clone --depth 1 https://aur.archlinux.org/sope.git &&  chown -R build ./sope
+WORKDIR /build/sope
+RUN sudo -u build makepkg -is --noconfirm && rm -rf /build/sope && yes | pacman -Sccq
+
+WORKDIR /build
+RUN git clone --depth 1 https://aur.archlinux.org/sogo.git &&  chown -R build ./sogo
```
Comment (DavidePrincipi) [inline Dockerfile:17]: Should we check out a fixed SOGo version number? Like 5.9.0... Same applies to its dependencies, Sope and libwxml.

As alternative to git clone, is it possible to download a tarball?

**Comment by stephdl** [comment]
@DavidePrincipi 

we require a specific version of archlinux
git has been removed, it slightly lower the image footprint, curl is used
 
 however I tried to install a specific version that we could require, with yay as an alternative but it does not offer to install a specific version too and it brings a lot more of dependencies like git and go.
 
So at this point we could download the build files and bundle them here. Like this we could sed the version we want but we need as a developer compare what we could miss for a future release

**Comment by stephdl** [comment]
ready to review @DavidePrincipi

File: build-images.sh [hunk 4]
```diff
@@ -7,56 +7,96 @@
 
 # Terminate on error
 set -e
-
+archlinux_version=base-20231112.0.191179
```
Comment (DavidePrincipi) [inline build-images.sh]:
```suggestion
archlinux_version=latest
```

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**